go/pools/smartconnpool/pool.go

```diff
@@ -161,7 +161,6 @@ type ConnPool[C Connection] struct {
 // The pool must be ConnPool.Open before it can start giving out connections
 func NewPool[C Connection](config *Config[C]) *ConnPool[C] {
 	pool := &ConnPool[C]{}
-	pool.freshSettingsStack.Store(-1)
```

Comment:
Was -1 being used as some kind of sentinel value? Or was this completely unnecessary?

Reply:
It was a sentinel value to mark when no Setting connection had been returned to the pool yet, but it was not an interesting optimization, so I removed it.